common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -67,13 +48,18 @@ public static String unquoteText(String text) {
     for (int chIndex = 1; chIndex < text.length() - 1; chIndex++) {
       currentChar = text.charAt(chIndex);
       nextChar = text.charAt(chIndex + 1);
-      if (currentChar == enclosingQuote
-          && nextChar == currentChar) {
+
+      if ((currentChar == '\\'
```

Comment:
logic here is very hard to follow (containing both && and || and lots of brackets. I'd suggest simplifying it. 

May have one if statement for `currentChar == '\\'` and another for `currentChar == enclosingQuote`

Reply:
Definitely need to put a set of brackets around 
```
currentChar == nextChar
          && currentChar == enclosingQuote
```